src/dvc_studio_client/post_live_metrics.py

```diff
@@ -68,6 +71,86 @@ def get_studio_token_and_repo_url(studio_token=None, studio_repo_url=None):
     return studio_token, studio_repo_url
 
 
+def get_studio_config(
+    dvc_config: Optional[Dict[str, Any]] = None,
+    offline: bool = False,
+    studio_token: Optional[str] = None,
+    studio_repo_url: Optional[str] = None,
+    studio_url: Optional[str] = None,
+) -> Dict:
+    """Get studio config options.
+
+    Args:
+        dvc_config (Optional[dict]): Dict returned by dvc.Repo.config["studio"].
+        offline (bool): Whether offline mode is enabled. Default: false.
+        studio_token (Optional[str]): Studio access token obtained from the UI.
+        studio_repo_url (Optional[str]): URL of the Git repository that has been
+            imported into Studio UI.
+        studio_url (Optional[str]): Base URL of Studio UI (if self-hosted).
+    Returns:
+        Dict:
+            Config options for posting live metrics.
+            Keys match the DVC_STUDIO... environment variables.
+    """
+
+    config = {}
+    if not dvc_config:
+        dvc_config = {}
+
+    def to_bool(var):
+        if var is None:
+            return False
+        return bool(re.search("1|y|yes|true", var, flags=re.I))
+
+    offline = (
+        offline
+        or to_bool(getenv(DVC_STUDIO_OFFLINE))
+        or to_bool(dvc_config.get("offline"))
+    )
+    if offline:
+        logger.debug("Offline mode enabled. Skipping `post_studio_live_metrics`")
+        return {}
+
+    studio_token = (
+        studio_token
+        or getenv(DVC_STUDIO_TOKEN)
+        or getenv(STUDIO_TOKEN)
+        or dvc_config.get("token")
+    )
+    if studio_token:
+        config["studio_token"] = studio_token
+    else:
+        logger.debug(
+            f"{DVC_STUDIO_TOKEN} not found. Skipping `post_studio_live_metrics`"
+        )
+        return {}
```

Comment:
Just my  personal preference, feel free to disregard

```suggestion
    if not studio_token:
        logger.debug(
            f"{DVC_STUDIO_TOKEN} not found. Skipping `post_studio_live_metrics`"
        )
        return {}
     config["studio_token"] = studio_token
```